app/scrapers/sections_scraper.rb

```diff
@@ -35,18 +35,29 @@
 # Parse section data from pages
 section_queries.each do |query|
   semester = query.scan(/soc\/(.+)\//)[0][0]
-  coll = db.collection("sections#{semester}")
-  bulk = coll.initialize_unordered_bulk_op
+  sections_coll = db.collection("sections#{semester}")
+  prof_coll = db.collection("profs#{semester}")
+  sections_bulk = sections_coll.initialize_unordered_bulk_op
+  prof_bulk = prof_coll.initialize_unordered_bulk_op
   page = Nokogiri::HTML(open(query))
   course_divs = page.search("div.course-sections")
   section_array = []
+  profs = {} # hash of professor => array of courses
 
   # for each of the courses on the page
   course_divs.each do |course_div|
     course_id = course_div.attr('id')
     # for each section of the course
     course_div.search("div.section").each do |section|
       # add section to array to add
+      instructors = section.search('span.section-instructors').text.gsub(/\t|\r\n/,'').encode('UTF-8', :invalid => :replace).split(',').map(&:strip)
+      
+      # add instructor to profs object, add course to prof's course set
+      instructors.each do |x| 
+        profs[x] ||= []
+        profs[x] |= [course_id] # adds iff course_id not in profs[x]
```

Comment:
This is a way to add an element to the array only if it doesn't already exist. Basically returns the union of bits in `profs[x]` and `[course_id]`, which will just be `profs[x]` if `course_id` already exists in the array.